pkg/kubelet/dockertools/labels.go

```diff
@@ -190,23 +204,64 @@ func getInt64PointerFromLabel(labels map[string]string, label string) (*int64, e
 func getJsonObjectFromLabel(labels map[string]string, label string, value interface{}) (bool, error) {
 	if strValue, found := labels[label]; found {
 		err := json.Unmarshal([]byte(strValue), value)
+		if err != nil {
+			err = parseLabelError(label, strValue, err)
+		}
 		return found, err
 	}
 	// Because it's normal that a container has no PreStopHandler label, don't report any error here.
 	return false, nil
 }
 
-// The label kubernetesPodLabel is added a long time ago (#7421), it serialized the whole api.Pod to a docker label.
+// Both kubernetes and docker recommend to only use [a-z0-9-.] in label keys, so we decide to replace the old labels.
+// The old labels are added before v1.2, and should be completely removed after we drop support for v1.2.
+// TODO(random-liu): Remove this function when we drop support for v1.2.
+func supplyContainerInfoWithV12Label(labels map[string]string, containerInfo *labelledContainerInfo) {
+	var err error
+	// Only try to get information when there are old labels so as to avoid unnecessary error message.
+	if _, found := labels[oldPodDeletionGracePeriodLabel]; found {
+		containerInfo.PodDeletionGracePeriod, err = getInt64PointerFromLabel(labels, oldPodDeletionGracePeriodLabel)
+		if err != nil {
+			logLabelError(*containerInfo, oldPodDeletionGracePeriodLabel, err)
+		}
+	}
+	if _, found := labels[oldPodTerminationGracePeriodLabel]; found {
+		containerInfo.PodTerminationGracePeriod, err = getInt64PointerFromLabel(labels, oldPodTerminationGracePeriodLabel)
```

Comment:
yes, it won't happen now but may happen in the future :)
and I raised this question also because your tests seemed to be testing the cases where half of the labels are new and the others were old.